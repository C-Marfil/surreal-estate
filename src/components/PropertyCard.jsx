import React from "react";
import PropTypes from "prop-types";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  const mailto = `mailto:${email}`;

  return (
    <div>
      <p>{title}</p>
      <p>{`${type}, ${city}`}</p>
      <ul>
        <li>bathroom:{bathrooms}</li>
        <li>bedrooms: {bedrooms}</li>
        <li>moneys:${price}</li>
      </ul>
      <p>
        <a href={mailto}>Email</a>
      </p>
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};

export default PropertyCard;
