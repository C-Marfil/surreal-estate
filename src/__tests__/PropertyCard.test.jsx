import React from "react";
import { getByRole, render, screen } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard element and its props", () => {
  it("renders the component within another component", () => {
    const { asFragment } = render(<PropertyCard />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the props correctly", () => {
    render(
      <PropertyCard
        type="flat"
        title="Resident Evil"
        bathrooms="2"
        bedrooms="2"
        email="hey@gmail.com"
        price="2000"
        city="Manchester"
      />
    );

    const list = screen.getByRole("list");

    expect(list).toBeInTheDocument();
  });
});
