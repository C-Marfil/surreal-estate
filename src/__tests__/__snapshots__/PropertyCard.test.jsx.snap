// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard element and its props renders the component within another component 1`] = `
<DocumentFragment>
  <div>
    <p>
      I am the property card
    </p>
    <p />
    <p />
    <ul>
      <li>
        bathroom:
      </li>
      <li>
        bedrooms:
      </li>
      <li>
        moneys:$
      </li>
    </ul>
    <p>
      <a
        href="mailto:undefined"
      >
        Email
      </a>
    </p>
  </div>
</DocumentFragment>
`;
